### PR TITLE
KTOR-7586: improve parsing of supported media types

### PR DESCRIPTION
**Pull request by anton-erofeev**
**Subsystem**
Core, http

**Motivation**
https://youtrack.jetbrains.com/issue/KTOR-7586

**Solution**
improved parsing of supported media types

### CODE REVIEW

**Review by osipxd — CHANGES_REQUESTED** [review]
Thank you for your contribution! 🚀 
I have some ideas on how to improve it. Please, check the comments.

File: ktor-http/common/src/io/ktor/http/Mimes.kt [hunk 1]
```diff
@@ -8,1231 +8,1000 @@ import io.ktor.util.*
...
+application/iff,iff
+application/iges,iges igs
+application/inf,inf
+application/ipfix,ipfix
+application/java,class
+application/java-archive,jar
+application/java-byte-code,class
+application/java-serialized-object,ser
+application/java-vm,class
+application/javascript,js
```
[the remark this answers is not in the document]
Comment (anton-erofeev) [reply ktor-http/common/src/io/ktor/http/Mimes.kt:64]: > after some separator

are you suggesting to insert a comment into `rawMimes` and ignore it when processing the string?

[the remark this answers is not in the document]
Comment (osipxd) [reply ktor-http/common/src/io/ktor/http/Mimes.kt:64]: Yes, we could ignore lines starting with `#` in addition to already ignored empty lines

**Comment by anton-erofeev** [comment]
> Thank you for your contribution! 🚀 I have some ideas on how to improve it. Please, check the comments.

good points, thanks, please review the fixes

**Review by osipxd — APPROVED** [review]
Thank you, this is a great job!
Just a few minor comments left

**Comment by osipxd** [comment]
Than you, I'll merge this PR to 3.1.0-eap 🎉

**Pull request merged**